src/main/java/jdraw/figures/FigureBase.java

```diff
@@ -7,10 +7,19 @@
 
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.stream.Collectors;
 
 public abstract class FigureBase implements Figure {
+    private final List<FigureListener> changeListeners;
+
+    public FigureBase(FigureBase figureBase) {
+        this.changeListeners = new CopyOnWriteArrayList<>(figureBase.changeListeners);
+    }
```

Comment:
i usually go for the copy constructor, i think this is more explicit. But it depends on the use case. two different approaches mixed together is not nice though. thanks for pointing this out

Reply:
We are discussing to drop the `Cloneable` interface, because the real disadvantage I see is that fields can't be declared final. And yes, the copy-constructor approach is a general approach which can be used in other languages as well, so yes, probably we will go into this direction as well.

Btw, the problem with the listeners would be solved correctly automatically with the copy constructor approach, because this list is initialized for new figures, i.e. without any special handling you get an empty list in your cloned instances.

> i usually go for the copy constructor

:+1: 
so continue with this approach.